compile-time warnings and possibly faulty test

There are several warnings with gcc 11 and -Wall at the moment.  They are mostly straightforward to fix, but one of them is confusing:

```
../tests/my-rpc.c: In function ‘my_rpc_hang_ult’:
../tests/my-rpc.c:153:54: warning: ‘hgi’ may be used uninitialized [-Wmaybe-uninitialized]
  153 |     hret = margo_bulk_transfer(mid, HG_BULK_PULL, hgi->addr, in.bulk_handle, 0,
      |                                                   ~~~^~~~~~
```

The `hgi` variable is indeed not initialized; it was probably cut in the recent `HG_Get_data()` changes.  But if it isn't initialized, how is the make check test case working if the above call relies on `hgi->addr`?

```
PASS: tests/timeout.sh
```
I started to fix the code but I'd like to understand how the test case is passing first to make sure there isn't another problem with this test.

--- a/src/margo-core.c
+++ b/src/margo-core.c
@@ -1076,9 +1076,8 @@ margo_irespond_internal(hg_handle_t   handle,
                         void*         out_struct,
                         margo_request req) /* should have been allocated */
 {
-    int                   ret;
-    hg_proc_cb_t          out_cb = NULL;
-    const struct hg_info* hgi    = HG_Get_info(handle);
+    int          ret;
+    hg_proc_cb_t out_cb = NULL;
 
     struct margo_handle_data* handle_data
         = (struct margo_handle_data*)HG_Get_data(handle);
@@ -1360,8 +1359,8 @@ hg_return_t margo_bulk_transfer(margo_instance_id mid,
 {
     struct margo_request_struct reqs;
     hg_return_t                 hret = margo_bulk_itransfer_internal(
-        mid, op, origin_addr, origin_handle, origin_offset, local_handle,
-        local_offset, size, &reqs);
+                        mid, op, origin_addr, origin_handle, origin_offset, local_handle,
+                        local_offset, size, &reqs);
     if (hret != HG_SUCCESS) return hret;
     return margo_wait_internal(&reqs);
 }

--- a/src/margo-diag.c
+++ b/src/margo-diag.c
@@ -200,8 +200,6 @@ void __margo_breadcrumb_measure(margo_instance_id     mid,
     size_t                    s, s1;
     struct margo_handle_data* margo_data;
     if (type) {
-        const struct hg_info* info;
-        info       = HG_Get_info(h);
         margo_data = (struct margo_handle_data*)HG_Get_data(h);
         if (margo_data && margo_data->pool != ABT_POOL_NULL) {
             ABT_pool_get_total_size(margo_data->pool, &s);

--- a/tests/my-rpc.c
+++ b/tests/my-rpc.c
@@ -132,7 +132,9 @@ static void my_rpc_hang_ult(hg_handle_t handle)
            in.input_val);
     out.ret = 0;
 
-    /* get margo instance */
+    /* get handle info and margo instance */
+    hgi = margo_get_info(handle);
+    assert(hgi);
     mid = margo_hg_handle_get_instance(handle);
     assert(mid != MARGO_INSTANCE_NULL);
 

--- a/tests/my-rpc.c
+++ b/tests/my-rpc.c
@@ -136,8 +136,8 @@ static void my_rpc_hang_ult(hg_handle_t handle)
     mid = margo_hg_handle_get_instance(handle);
     assert(mid != MARGO_INSTANCE_NULL);
 
-    /* sleep for an hour (to allow client to test timeout capability) */
-    margo_thread_sleep(mid, 1000 * 60 * 60);
+    /* sleep for 5 seconds (to allow client to test timeout capability) */
+    margo_thread_sleep(mid, 5000);
 
     /* set up target buffer for bulk transfer */
     size   = 512;

--- a/tests/unit-tests/margo-bulk.c
+++ b/tests/unit-tests/margo-bulk.c
@@ -106,6 +106,7 @@ static MunitResult test_margo_bulk_create(const MunitParameter params[],
 error:
     margo_bulk_free(bulk_handle);
     free(buffer);
+    return MUNIT_FAIL;
 }
 
 static char* protocol_params[] = {"na+sm", NULL};

--- a/tests/unit-tests/margo-comm-finalize.c
+++ b/tests/unit-tests/margo-comm-finalize.c
@@ -27,18 +27,31 @@ static void test_rpc_ult(hg_handle_t handle)
     margo_instance_id mid = MARGO_INSTANCE_NULL;
 
     mid = margo_hg_handle_get_instance(handle);
-    /* munit_assert_not_null(mid); */
+    if(!mid) {
+        munit_error("margo_hg_handle_get_instance() failure");
+        return;
+    }
 
     hret = margo_get_input(handle, &in);
-    /* munit_assert_int(hret, ==, HG_SUCCESS); */
+    if(hret != HG_SUCCESS) {
+        munit_error("margo_get_input() failure");
+        return;
+    }
 
     if(in.dereg_flag) {
         hret = margo_deregister(mid, test_rpc_id);
-        /* munit_assert_int(hret, ==, HG_SUCCESS); */
+        if(hret != HG_SUCCESS) {
+            munit_error("margo_deregister() failure");
+            return;
+        }
     } else {
         // it is undefined behavior to call margo_respond
         // after the RPC has been deregistered
         hret = margo_respond(handle, NULL);
+        if(hret != HG_SUCCESS) {
+            munit_error("margo_respond() failure");
+            return;
+        }
     }
 
     margo_destroy(handle);

--- a/tests/unit-tests/margo-forward.c
+++ b/tests/unit-tests/margo-forward.c
@@ -95,7 +95,6 @@ cleanup:
 
     hret[4] = margo_addr_free(ctx->mid, addr);
 
-check:
     munit_assert_int_goto(hret[0], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[1], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[2], ==, HG_SUCCESS, error);
@@ -183,7 +182,6 @@ cleanup:
 
     hret[4] = margo_addr_free(ctx->mid, addr);
 
-check:
     munit_assert_int_goto(hret[0], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[1], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[2], ==, HG_NO_MATCH, error);
@@ -222,7 +220,6 @@ cleanup:
     hret[3] = margo_destroy(handle);
     hret[4] = margo_addr_free(ctx->mid, addr);
 
-check:
     munit_assert_int_goto(hret[0], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[1], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[2], ==, HG_NO_MATCH, error);
@@ -261,7 +258,6 @@ cleanup:
     hret[3] = margo_destroy(handle);
     hret[4] = margo_addr_free(ctx->mid, addr);
 
-check:
     munit_assert_int_goto(hret[0], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[1], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[2], ==, HG_NO_MATCH, error);
@@ -300,7 +296,6 @@ cleanup:
     hret[3] = margo_destroy(handle);
     hret[4] = margo_addr_free(ctx->mid, addr);
 
-check:
     munit_assert_int_goto(hret[0], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[1], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[2], ==, HG_SUCCESS, error);
@@ -340,7 +335,6 @@ cleanup:
     hret[3] = margo_destroy(handle);
     hret[4] = margo_addr_free(ctx->mid, addr);
 
-check:
     munit_assert_int_goto(hret[0], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[1], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[2], ==, HG_NO_MATCH, error);
@@ -381,7 +375,6 @@ cleanup:
     hret[3] = margo_destroy(handle);
     hret[4] = margo_addr_free(ctx->mid, addr);
 
-check:
     munit_assert_int_goto(hret[0], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[1], ==, HG_SUCCESS, error);
     munit_assert_int_goto(hret[2], ==, HG_NO_MATCH, error);

